src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/FileStorageService.cs

```diff
@@ -160,31 +157,38 @@ public async Task DeletePublicBlob(string path)
             await _publicBlobStorageService.DeleteBlob(PublicContentContainerName, path);
         }
 
-        public async Task<IEnumerable<FileInfo>> ListPublicFiles(string publication, string release)
+        [Obsolete("Use GetPublicFileInfo with a file reference instead of filename")]
+        public async Task<FileInfo> GetPublicFileInfo(ReleaseFileTypes type, string path)
         {
-            var files = new List<FileInfo>();
-
-            files.AddRange(
-                (await _publicBlobStorageService.ListBlobs(
-                    PublicFilesContainerName,
-                    PublicReleaseDirectoryPath(publication, release, ReleaseFileTypes.Data)
-                ))
-                .Where(blob => !blob.IsMetaDataFile() && blob.IsReleased())
-                .Select(blob => blob.ToFileInfo(ReleaseFileTypes.Data))
-            );
-            files.AddRange(
-                (await _publicBlobStorageService.ListBlobs(
-                    PublicFilesContainerName,
-                    PublicReleaseDirectoryPath(publication, release, ReleaseFileTypes.Ancillary)
-                ))
-                .Where(blob => blob.IsReleased())
-                .Select(blob => blob.ToFileInfo(ReleaseFileTypes.Ancillary))
+            var blob = await _publicBlobStorageService.GetBlob(PublicFilesContainerName, path);
 
-            );
+            return new FileInfo
+            {
+                Id = null,
+                FileName = blob.FileName,
+                Name = blob.Name,
+                Path = blob.Path,
+                Size = blob.Size,
+                Type = type
+            };
+        }
+
+        public async Task<FileInfo> GetPublicFileInfo(string publication, string release, ReleaseFileReference file)
```

Comment:
Should be able to get to the publication and release slugs from the file's `Release.Publication`. I'm not too keen about passing in the slugs separately, as it could obviously lead to incorrect usage. Having only the `file` ensure you can't mis-use it (and is also less confusing). However, I'm also conscious that it puts the onus on the consumer to correctly hydrate the `Release.Publication` relationship.

IMO, I would suggest exposing the `CheckBlobExists` and `GetBlob` storage methods (these are much more userful) and then just work with those in `ReleaseService`. I'm not sure there's much utility in this being a method on this service.

Reply:
It's not the same as the `ReleaseFileReference.Release` and `ReleaseFileReference.Release.Publication` because we always publish files on the slugs of the Release being published. 

`ReleaseFileReference.Release` is the `Release` that the file was originally uploaded against.

Reply:
We could query for the `ReleaseFile` instead to get the right release thought right?

Reply:
I've moved the code to  `ReleaseService#GetPublicFileInfo` and use a new exposed `CheckBlobExists` and `GetBlob` like you suggested. `GetPublicFileInfo` method takes `Release` as a parameter now which is the correct Release.

Reply:
Made a further change in `FileExtensions` to not require separate slugs to get the public path, but use a Release parameter instead which cleans things up nicely. Also if you have a `ReleaseFile` rather than a `ReleaseFileReference` then we assume you can use the `Release` from that and it doesn't need a separate `Release`.